crates/sui-verifier/src/private_generics.rs

```diff
@@ -26,10 +28,12 @@ use crate::{format_signature_token, verification_failure};
 /// Concretely, with `event::emit<T>(...)`:
 /// - `T` must be a type declared in the current module
 pub fn verify_module(module: &CompiledModule) -> Result<(), ExecutionError> {
+    let test_scenario_module = *module.address() == SUI_FRAMEWORK_ADDRESS
+        && module.name() == IdentStr::new(TEST_SCENARIO_MODULE_NAME).unwrap();
```

Comment:
Can we just return here if this is true?

Reply:
We could! I went even more fine-grained here that was necessary probably and wanted to do it just for private transfers rather than both for private transfers and private emits.